internal/web/common/endpoint.go

```diff
@@ -0,0 +1,95 @@
+package common
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/eurofurence/reg-room-service/internal/apierrors"
+	"github.com/eurofurence/reg-room-service/internal/logging"
+)
+
+type (
+	RequestHandler[Req any]  func(r *http.Request) (*Req, error)
+	ResponseHandler[Res any] func(ctx context.Context, res *Res, w http.ResponseWriter) error
+	Endpoint[Req, Res any]   func(ctx context.Context, request *Req, logger logging.Logger) (*Res, error)
+)
+
+func CreateHandler[Req, Res any](endpoint Endpoint[Req, Res],
+	requestHandler RequestHandler[Req],
+	responseHandler ResponseHandler[Res],
+) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		reqID := logging.GetRequestID(ctx)
+		logger := logging.LoggerFromContext(ctx)
+
+		defer func() {
+			err := r.Body.Close()
+			if err != nil {
+				logger.Error("Error when closing the request body. [error]: %v", err)
+			}
+		}()
+
+		if requestHandler == nil {
```

Comment:
We can discuss that again, I initially thought so as well, just didn't like the code control flow then. Setting up routes would become a bit tedious with a lot of error checking for every single endpoint.

Let's talk about that again at a later time

Reply:
Honestly, if the routes can't be set up, it could just fatal out at startup time I guess. The service won't be working correctly anyway, and it's a programming error that will hopefully happen very rarely.